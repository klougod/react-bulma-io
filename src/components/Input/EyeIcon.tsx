import React from 'react'
import { faEye, faEyeSlash } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

interface EyeIconProps {
  showPw: Boolean
  onClick: () => void
}

export const EyeIcon = ({ showPw, onClick }: EyeIconProps) => (
  <p className='control'>
    <button className='button' tabIndex={-1} onClick={onClick}>
      <FontAwesomeIcon icon={showPw ? faEyeSlash : faEye} />
    </button>
  </p>
)

EyeIcon.defaultProps = {
  showPw: false,
  onClick: () => {}
}

export default EyeIcon
