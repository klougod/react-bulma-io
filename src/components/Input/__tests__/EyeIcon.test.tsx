import React from 'react'
import Enzyme from 'enzyme'

import EyeIcon from '../EyeIcon'

describe('EyeIcon component test', function () {
  it('should render without crash without props', function () {
    const inputMock = Enzyme.shallow(<EyeIcon />)
    expect(inputMock).toMatchSnapshot()
  })
})
