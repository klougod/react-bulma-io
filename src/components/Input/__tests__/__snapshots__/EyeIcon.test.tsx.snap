// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EyeIcon component test should render without crash without props 1`] = `
<p
  className="control"
>
  <button
    className="button"
    onClick={[Function]}
    tabIndex={-1}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f06e",
            "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
          ],
          "iconName": "eye",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</p>
`;
